lib/Composer.php

```diff
@@ -100,6 +101,52 @@ public static function executeMigrationsUp(Event $event)
         }
     }
 
+
+    /**
+     * @param Event $event
+     */
+    public static function executeBundleMigrationsUp(Event $event)
+    {
+        $consoleDir = static::getConsoleDir($event, 'bundle migrations');
+
+        if (null === $consoleDir) {
+            return;
+        }
+
+        $bundles = include('var/config/extensions.php'); // path is based on composer.json's path
+
+        $installedBundles = array_map(static function($bundleClassName, $active) {
+            if($active) {
+                return substr($bundleClassName, strrpos($bundleClassName, "\\")+1);
+            }
+            return null;
+        }, array_keys($bundles['bundle']), $bundles['bundle']);
+
+        $installedBundles = array_filter($installedBundles);
```

Comment:
Would be easier if you filter first:

```suggestion
        $installedBundles = array_filter($bundles['bundle']);
        $installedBundles = array_map(static function($bundleClassName) {
            return substr($bundleClassName, strrpos($bundleClassName, "\\") + 1);
        }, array_keys($installedBundles));
```

---
I'd also suggest that you use reflection to get the short name of the class, as it's almost as fast, but expresses the intention much better:

```suggestion
        $installedBundles = array_map(static function($bundleClassName) {
            return (new \ReflectionClass($bundleClassName))->getShortName();
        }, array_keys(array_filter($bundles['bundle'])));
```

Reply:
Filtering first does not work because this array_map sets array items to `null` if the respective bundles are disabled. The `array_filter` thereafter does remove those items.

Now `ReflectionClass` gets used to get the bundle class' short name, thanks.

Reply:
That would be the exact advantage of filtering first: then there's no need to set it to `null` because only active bundles (those with a value of `true`) arrive into `array_map` ;)

Reply:
Oh you are right, sorry. Changed it.